tests/control/ros/test_Conversions.cpp

```diff
@@ -0,0 +1,275 @@
+  EXPECT_EIGEN_EQUAL(make_vector(0.), values, kTolerance);
+}
+
+TEST_F(ConvertJointTrajectoryTests, CubicTrajectory)
+{
+  auto cubicTwoWaypointMessage = mTwoWaypointMessage;
+  cubicTwoWaypointMessage.points[0].velocities.assign({3.});
+  cubicTwoWaypointMessage.points[1].velocities.assign({4.});
+
+  const auto trajectory = convertJointTrajectory(
+    mStateSpace, cubicTwoWaypointMessage);
+
+  ASSERT_TRUE(!!trajectory);
+  ASSERT_DOUBLE_EQ(0., trajectory->getStartTime());
+  ASSERT_DOUBLE_EQ(1., trajectory->getEndTime());
+  EXPECT_DOUBLE_EQ(1., trajectory->getDuration());
+  EXPECT_EQ(1, trajectory->getNumSegments());
+  EXPECT_EQ(3, trajectory->getNumDerivatives());
+
+  auto state = mStateSpace->createState();
+  Eigen::VectorXd values;
+
+  trajectory->evaluate(0., state);
+  mStateSpace->convertStateToPositions(state, values);
+  EXPECT_EIGEN_EQUAL(make_vector(1.), values, kTolerance);
+
+  trajectory->evaluate(1., state);
+  mStateSpace->convertStateToPositions(state, values);
+  EXPECT_EIGEN_EQUAL(make_vector(2.), values, kTolerance);
+
+  trajectory->evaluateDerivative(0., 1, values);
+  EXPECT_EIGEN_EQUAL(make_vector(3.), values, kTolerance);
+
+  trajectory->evaluateDerivative(1., 1, values);
+  EXPECT_EIGEN_EQUAL(make_vector(4.), values, kTolerance);
+}
+
+TEST_F(ConvertJointTrajectoryTests, QuinticTrajectory)
+{
+  auto quinticTwoWaypointMessage = mTwoWaypointMessage;
+  quinticTwoWaypointMessage.points[0].velocities.assign({3.});
+  quinticTwoWaypointMessage.points[0].accelerations.assign({5.});
+  quinticTwoWaypointMessage.points[1].velocities.assign({4.});
+  quinticTwoWaypointMessage.points[1].accelerations.assign({6.});
+
+  const auto trajectory = convertJointTrajectory(
+    mStateSpace, quinticTwoWaypointMessage);
+
+  ASSERT_TRUE(!!trajectory);
+  ASSERT_DOUBLE_EQ(0., trajectory->getStartTime());
+  ASSERT_DOUBLE_EQ(1., trajectory->getEndTime());
+  EXPECT_DOUBLE_EQ(1., trajectory->getDuration());
+  EXPECT_EQ(1, trajectory->getNumSegments());
+  EXPECT_EQ(5, trajectory->getNumDerivatives());
+
+  auto state = mStateSpace->createState();
+  Eigen::VectorXd values;
+
+  trajectory->evaluate(0., state);
+  mStateSpace->convertStateToPositions(state, values);
+  EXPECT_EIGEN_EQUAL(make_vector(1.), values, kTolerance);
+
+  trajectory->evaluate(1., state);
+  mStateSpace->convertStateToPositions(state, values);
+  EXPECT_EIGEN_EQUAL(make_vector(2.), values, kTolerance);
+
+  trajectory->evaluateDerivative(0., 1, values);
+  EXPECT_EIGEN_EQUAL(make_vector(3.), values, kTolerance);
+
+  trajectory->evaluateDerivative(1., 1, values);
+  EXPECT_EIGEN_EQUAL(make_vector(4.), values, kTolerance);
+
+  trajectory->evaluateDerivative(0., 2, values);
+  EXPECT_EIGEN_EQUAL(make_vector(5.), values, kTolerance);
+
+  trajectory->evaluateDerivative(1., 2, values);
+  EXPECT_EIGEN_EQUAL(make_vector(6.), values, kTolerance);
+}
+
+TEST_F(ConvertJointTrajectoryTests, LinearTrajectoryWithDifferentOrdering)
```

Comment:
Nit: The fact that you are using a linear trajectory in this test is an implementation detail. I suggest renaming the test to just `DifferentJointOrdering`.